From aaa19ebf42f6bc9f82f34985f956afe447e72c8c Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 24 Jul 2014 09:57:17 +0800
Subject: [PATCH] make_ext.pl: fix regenerate makefile failed while $cc changed

While $cc changed, the existance of 'xdefine' caused makefile
regeneration failed.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

---
 make_ext.pl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/make_ext.pl b/make_ext.pl
index 80d8f68..823efe0 100644
--- a/make_ext.pl
+++ b/make_ext.pl
@@ -336,6 +336,10 @@ sub build_extension {
                 print "Deleting non-Cross makefile\n";
                 close $mfh or die "close $makefile: $!";
                 _unlink($makefile);
+                if(-e 'xdefine') {
+                    print "Deleting xdefine for regenerate makefile\n";
+                    _unlink('xdefine');
+                }
             }
         }
     } else {
